File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/fragment/FragmentModule.java
package com.elveselimoski.starterapp.di.fragment;

import androidx.fragment.app.Fragment;
import com.elveselimoski.starterapp.di.scope.PerFragment;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Elvedin Selimoski on 12/6/17. mail: elveselimoski@gmail.com
 */

@Module
public class FragmentModule {
    private Fragment fragment;

    public FragmentModule(Fragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    @PerFragment
    Fragment fragment() {
        return this.fragment;
    }

}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/fragment/FragmentModule.kt
package com.elveselimoski.starterapp.di.fragment

import androidx.fragment.app.Fragment
import com.elveselimoski.starterapp.di.scope.PerFragment
import dagger.Module
import dagger.Provides

/**
 * Created by Elvedin Selimoski on 12/6/17. mail: elveselimoski@gmail.com
 */

@Module
class FragmentModule(private val fragment: Fragment) {

    @Provides
    @PerFragment
    internal fun fragment(): Fragment {
        return this.fragment
    }

}
